src/fs.cpp

```diff
@@ -52,15 +55,26 @@ bool FileLock::TryLock()
     if (fd == -1) {
         return false;
     }
-    struct flock lock;
-    lock.l_type = F_WRLCK;
-    lock.l_whence = SEEK_SET;
-    lock.l_start = 0;
-    lock.l_len = 0;
-    if (fcntl(fd, F_SETLK, &lock) == -1) {
-        reason = GetErrorReason();
-        return false;
+    // Exclusive file locking is broken on WSL using fcntl (issue #18622)
+    // This workaround can be removed once the bug on WSL is fixed
+    struct utsname uname_data;
+    if (uname(&uname_data) == 0 && std::string(uname_data.version).find("Microsoft") != std::string::npos) {
```

Comment:
This condition should probably be cached in a static bool local variable.

Reply:
@sipa it isn't "win32" because its the linux subsystem, so it runs in the non-windows branch

Reply:
Ugh.

Reply:
Still, please cache the result.